packages/expo-cli/src/commands/build/AndroidBuilder.js

```diff
@@ -212,11 +212,20 @@ export default class AndroidBuilder extends BaseBuilder {
   async validateProject(options) {
     const publicUrl = options.publicUrl;
     const {
-      args: { sdkVersion },
+      args: { sdkVersion, androidPackage },
     } = publicUrl
-      ? await Exp.getThirdPartyInfoAsync(publicUrl)
-      : await Exp.getPublishInfoAsync(this.projectDir);
+      ? await Exp.getThirdPartyInfoAsync(publicUrl, 'android')
+      : await Exp.getPublishInfoAsync(this.projectDir, 'android');
     await this.checkIfSdkIsSupported(sdkVersion, 'android');
+
+    if (!androidPackage) {
+      throw new XDLError(
+        ErrorCode.INVALID_OPTIONS,
+        `Your project must have an Android package set in app.json.
```

Comment:
The error message should mention the field is `android.package`, or even better, prompt the user for the package name and write it to `app.json`, example: https://github.com/expo/expo-cli/blob/371be04175a676ae789544836ada6421a5d95937/packages/xdl/src/detach/Detach.js#L213-L225

Reply:
updating a local android package would be confusing for users using self-hosted apps. We would update local app.json and deploy remote one.